### PR TITLE
fix issue running tests in docker

### PR DESCRIPTION
**Pull request by jluhrsen**
something has changed in the interaction between git and golang 1.18.5 and how it works with docker. It's not a problem in 1.18.4 and also not a problem running tests with podman.

the way we are launching the docker container, both git and go have some issue. Adding the root of the ovn-kubernetes as a "safe.directory" in the global git config will resolve it:

```
❯ docker run -it --rm -u root --security-opt label=disable --cap-add=NET_ADMIN --cap-add=SYS_ADMIN -v /home/jamoluhrsen/repos/RedHat/ovn-kubernetes:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.19.5 bash root@61900f0fd213:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# git status fatal: detected dubious ownership in repository at '/go/src/github.com/ovn-org/ovn-kubernetes' To add an exception for this directory, call:

        git config --global --add safe.directory /go/src/github.com/ovn-org/ovn-kubernetes
root@61900f0fd213:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# ./hack/test-go.sh go test -mod=vendor -test.v  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/clustermanager -ginkgo.v  -ginkgo.reportFile ./_artifacts/junit-pkg_clustermanager.xml ^C
root@61900f0fd213:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# go list ./cmd/... github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-k8s-cni-overlay github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-kube-util github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-kube-util/app github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovndbchecker github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovnkube github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovnkube-trace root@61900f0fd213:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# exit exit
❯ docker run -it --rm -u root --security-opt label=disable --cap-add=NET_ADMIN --cap-add=SYS_ADMIN -v /home/jamoluhrsen/repos/RedHat/ovn-kubernetes:/go/src/github.com/ovn-org/ovn-kubernetes -w /go/src/github.com/ovn-org/ovn-kubernetes/go-controller -e COVERALLS= -e GINKGO_FOCUS="" quay.io/giantswarm/golang:1.19.5 bash root@f4e9245e2268:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# go list ./cmd/... error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
root@f4e9245e2268:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# git status fatal: detected dubious ownership in repository at '/go/src/github.com/ovn-org/ovn-kubernetes' To add an exception for this directory, call:

        git config --global --add safe.directory /go/src/github.com/ovn-org/ovn-kubernetes
root@f4e9245e2268:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# git config --global --add safe.directory /go/src/github.com/ovn-org/ovn-kubernetes root@f4e9245e2268:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# go list ./cmd/... github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-k8s-cni-overlay github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-kube-util github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-kube-util/app github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovndbchecker github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovnkube github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovnkube-trace root@f4e9245e2268:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# git status On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   hack/test-go.sh

no changes added to commit (use "git add" and/or "git commit -a") root@f4e9245e2268:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller#
```

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/hack/test-go.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ setup_env
 
 cd "${OVN_KUBE_ROOT}"
 
+git config --global --add safe.directory "$( absolute_path "${OVN_KUBE_ROOT}/.." )"
```
Comment (flavio-fernandes) [inline go-controller/hack/test-go.sh:11]: 👍

**Comment by jcaamano** [comment]
nit: isn't `-buildvcs=false` the best option to use here?

**Comment by jluhrsen** [comment]
> nit: isn't `-buildvcs=false` the best option to use here?

TBH, I had no idea if the VCS info being used in the build was important for anything or anybody so figured letting
it still work by flagging a "safe" directory was best. I really don't understand the gory details though, so if you think
`-buildvcs=false` is best I can change it. LMK

**Comment by jcaamano** [comment]
> > nit: isn't `-buildvcs=false` the best option to use here?
> 
> TBH, I had no idea if the VCS info being used in the build was important for anything or anybody so figured letting it still work by flagging a "safe" directory was best. I really don't understand the gory details though, so if you think `-buildvcs=false` is best I can change it. LMK

I don't think we are generating any consumable binary for which stamping VCS information would be useful for in the UTs.

I would try to use `-buildvcs=false` first because that has no security implications to think about. 

But not a strong opinion either. I don't think there are any security implications in the safety override. But you never know...

**Comment by jluhrsen** [comment]
> > > nit: isn't `-buildvcs=false` the best option to use here?
> > 
> > 
> > TBH, I had no idea if the VCS info being used in the build was important for anything or anybody so figured letting it still work by flagging a "safe" directory was best. I really don't understand the gory details though, so if you think `-buildvcs=false` is best I can change it. LMK
> 
> I don't think we are generating any consumable binary for which stamping VCS information would be useful for in the UTs.
> 
> I would try to use `-buildvcs=false` first because that has no security implications to think about.
> 
> But not a strong opinion either. I don't think there are any security implications in the safety override. But you never know...

Actually, I forgot that when I was narrowing this down it wasn't actually because of anything VCS and that
was just a side-effect error message we get from `go list...` The problem is something to do with docker
and go 1.19 and the files being owned by user 1000 which is not root.

even `git status` fails:

```
root@c84c8d14bdea:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# git status
fatal: detected dubious ownership in repository at '/go/src/github.com/ovn-org/ovn-kubernetes'
To add an exception for this directory, call:

        git config --global --add safe.directory /go/src/github.com/ovn-org/ovn-kubernetes
```

In this state, `go list...` gives us the VCS error:
```

root@c84c8d14bdea:/go/src/github.com/ovn-org/ovn-kubernetes/go-controller# go list ./cmd/...
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
```


I can get around this by doing a `chown -R $(whoami) ../.` instead of the safe.directory config, but
that has the hassle of changing the docker host's files to root user which has to be reverted by the
users afterward. I imagine that could cause a lot of headaches.

**Comment by jcaamano** [comment]
Adding ` -buildvcs=false` to that `go list` command worked for me.

All the go commands in that file probably require the flag and it might be worth it to just create a specific function for it. Somethign like this seemed to work for me:
```
diff --git a/go-controller/hack/test-go.sh b/go-controller/hack/test-go.sh
index 2fdb34fd7..5c9b8eda8 100755
--- a/go-controller/hack/test-go.sh
+++ b/go-controller/hack/test-go.sh
@@ -8,7 +8,13 @@ setup_env
 
 cd "${OVN_KUBE_ROOT}"
 
-PKGS=$(go list -mod vendor -f '{{if len .TestGoFiles}} {{.ImportPath}} {{end}}' ${PKGS:-./cmd/... ./pkg/... ./hybrid-overlay/...} | xargs)
+function gocmd {
+    cmd="go $1 -buildvcs=false"
+    shift
+    $cmd "$@"
+}
+
+PKGS=$(gocmd list -mod vendor -f "{{if len .TestGoFiles}} {{.ImportPath}} {{end}}" ${PKGS:-./cmd/... ./pkg/... ./hybrid-overlay/...} | xargs)
 
 if [[ "$1" == "focus" && "$2" != "" ]]; then
     ginkgo_focus="-ginkgo.focus="$(echo ${2} | sed 's/ /\\s/g')""
@@ -18,7 +24,7 @@ TEST_REPORT_DIR=${TEST_REPORT_DIR:="./_artifacts"}
 function testrun {
     local idx="${1}"
     local pkg="${2}"
-    local go_test="go test -mod=vendor"
+    local go_test="gocmd test -mod=vendor"
     local otherargs="${@:3} "
     local args=""
     local ginkgoargs=
@@ -31,9 +37,9 @@ function testrun {
         testfile=$(mktemp --tmpdir ovn-test.XXXXXXXX)
         echo "sudo required for ${pkg}, compiling test to ${testfile}"
         if [[ ! -z "${RACE:-}" ]]; then
-            go test -mod=vendor -race -covermode atomic -c "${pkg}" -o "${testfile}"
+            gocmd test -mod=vendor -race -covermode atomic -c "${pkg}" -o "${testfile}"
         else
-            go test -mod=vendor -covermode set -c "${pkg}" -o "${testfile}"
+            gocmd test -mod=vendor -covermode set -c "${pkg}" -o "${testfile}"
         fi
         args=""
         if [ "$NOROOT" = "TRUE" ]; then
@@ -58,7 +64,7 @@ function testrun {
         ginkgoargs="-ginkgo.v ${ginkgo_focus} -ginkgo.reportFile ${TEST_REPORT_DIR}/junit-${prefix}.xml"
     fi
     args="${args}${otherargs}"
-    if [ "$go_test" == "go test -mod=vendor" ]; then
+    if [ "$go_test" == "gocmd test -mod=vendor" ]; then
         args=${args}${pkg}
     fi
     echo "${go_test} -test.v ${args} ${ginkgoargs}"
```

**Comment by jluhrsen** [comment]
Thanks for the great advice @jcaamano. I've updated the PR with your recommendation.

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

**Review by jcaamano — APPROVED** [review]

**Pull request merged**